corehq/project_limits/tests/test_system_limit.py

```diff
@@ -22,3 +24,21 @@ def test_for_key_returns_domain_specific_limit(self):
         SystemLimit.objects.create(key="general_limit", limit=20, domain="specific")
         self.assertEqual(SystemLimit.for_key("general_limit"), 10)
         self.assertEqual(SystemLimit.for_key("general_limit", domain="specific"), 20)
+
+    def test_raises_error_if_changing_scope_from_global_to_domain(self):
+        global_limit = SystemLimit.objects.create(key="general_limit", limit=10)
+        with self.assertRaises(SystemLimitIllegalScopeChange):
+            global_limit.domain = 'new-domain'  # this is changing scope from global to domain
+            global_limit.save()
+
+        # nothing stops you from changing a domain scope to a global scope apart from the db constraint
+        new_limit = SystemLimit.objects.create(key="domain_limit", limit=5, domain='test')
+        new_limit.domain = ''
+        new_limit.save()
```

Comment:
These lines with `new_limit` seem like a separate test.

Reply:
Yeah I've seen others consolidate tests and was trying it out, but I'm all for it so added in 420b51d47cacf683bc38563ce64e15a1127bc06e.

Reply:
I do that occasionally when a single test does a sequence of setup steps that are expensive or involved enough that is annoying to split the test, but in general if it's always preferable to have separate tests for separate operations.